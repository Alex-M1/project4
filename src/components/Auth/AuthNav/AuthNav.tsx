import React from 'react';
import { NavLink } from 'react-router-dom';
import { useTranslation } from 'react-i18next';
import { CLIENT } from 'constants/urls';
import { useTheme } from 'src/components/hooks/useTheme';
import { StNavLink, StP } from '../styled';

const AuthNav: React.FC = () => {
  const { t } = useTranslation();
  const { colors, theme } = useTheme();
  return (
    <StNavLink
      theme={theme}
      colors={colors}
    >
      <p>{t('to_register')}</p>  
      &nbsp;
      <NavLink to={CLIENT.signUp}>
        <StP
          theme={theme}
          colors={colors}
        >
          {t('to_register_link')}
        </StP>
      </NavLink>
    </StNavLink>
  );
};

export default AuthNav;
