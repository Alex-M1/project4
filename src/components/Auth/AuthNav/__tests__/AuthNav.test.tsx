import React from 'react';
import AuthNav from '../AuthNav';
import { mountSmart, shallowSmart } from 'tests/testHelper';

jest.mock('react-i18next', () => ({
    useTranslation: jest.fn().mockImplementation(() => ({
        t: jest.fn(),
    })),
}),
);

describe('AuthNav', () => {
    it('Should match snapshot', () => {
        const component = shallowSmart(<AuthNav />);
        expect(component.html()).toMatchSnapshot();
    });
    it('should render StNavLink', () => {
        const component = mountSmart(<AuthNav />);
        expect(component.find('styled__StNavLink')).toHaveLength(1);
    });
    it('should change', () => {
        const component = mountSmart(<AuthNav />);
        expect(component.find('styled__StNavLink')).toHaveLength(1);
    });
});
