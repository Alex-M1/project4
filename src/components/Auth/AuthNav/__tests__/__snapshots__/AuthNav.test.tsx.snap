// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthNav Should match snapshot 1`] = `"<div class=\\"styled__StNavLink-kYzesN lhSwRU\\"><p></p>Â <a href=\\"/signup\\"><p class=\\"styled__StP-gOoQqg hCGbld\\"></p></a></div>"`;
